rpcs/ledgerService.go

```diff
@@ -229,24 +230,65 @@ func (ls *LedgerService) ListenForCatchupReq(reqs <-chan network.IncomingMessage
 // a blocking function for handling a catchup request
 func (ls *LedgerService) handleCatchupReq(ctx context.Context, reqMsg network.IncomingMessage) {
 	var res WsGetBlockOut
+	target := reqMsg.Sender.(network.UnicastPeer)
+	var respTopics network.Topics
+
+	if target.Version() == "1" {
+
+		defer func() {
+			ls.sendCatchupRes(ctx, target, reqMsg.Tag, res)
+		}()
+		var req WsGetBlockRequest
+		err := protocol.DecodeReflect(reqMsg.Data, &req)
+		if err != nil {
+			res.Error = err.Error()
+			return
+		}
+		res.Round = req.Round
+		encodedBlob, err := RawBlockBytes(ls.ledger, basics.Round(req.Round))
+
+		if err != nil {
+			res.Error = err.Error()
+			return
+		}
+		res.BlockBytes = encodedBlob
+		return
+	}
+	// Else, if version == 2.1
 	defer func() {
-		ls.sendCatchupRes(ctx, reqMsg.Sender.(network.UnicastPeer), reqMsg.Tag, res)
+		target.Respond(ctx, reqMsg, respTopics)
 	}()
 
-	var req WsGetBlockRequest
-	err := protocol.DecodeReflect(reqMsg.Data, &req)
+	topics, err := network.UnmarshallTopics(reqMsg.Data)
 	if err != nil {
-		res.Error = err.Error()
+		errMsg := "LedgerService handleCatchupReq UnmarshallTopics error: " + err.Error()
+		topics = network.Topics{
+			network.MakeTopic(network.ErrorKey, []byte(errMsg))}
 		return
 	}
-	res.Round = req.Round
-	encodedBlob, err := RawBlockBytes(ls.ledger, basics.Round(req.Round))
-
-	if err != nil {
-		res.Error = err.Error()
+	roundBytes, found := topics.GetValue(roundKey)
+	if !found {
+		topics = network.Topics{
+			network.MakeTopic(network.ErrorKey,
+				[]byte("LedgerService handleCatchupReq: round-number topic is missing"))}
 		return
 	}
-	res.BlockBytes = encodedBlob
+	requestType, found := topics.GetValue(requestDataTypeKey)
+	if !found {
+		topics = network.Topics{
+			network.MakeTopic(network.ErrorKey,
+				[]byte("LedgerService handleCatchupReq: request data-type is missing"))}
+		return
+	}
+	
+	round, read := binary.Uvarint(roundBytes)
+	if read <= 0 {
+		topics = network.Topics{
+			network.MakeTopic(network.ErrorKey,
+				[]byte("LedgerService handleCatchupReq: error reading the round number"))}
```

Comment:
while the error string here is valid, it's not really what the other party expects.
If you were to write this error to the local log, it would make sense. however, since you're sending it to the other side, you should omit the function name/service and focus on the reason this message was wrong. i.e. `Unable to parse round number`.